tools/mc-java-protoc/src/main/java/io/spine/tools/mc/java/protoc/field/FieldGenerationTask.java

```diff
@@ -27,34 +27,38 @@
 package io.spine.tools.mc.java.protoc.field;
 
 import com.google.common.collect.ImmutableList;
-import io.spine.tools.java.code.field.FieldFactory;
 import io.spine.code.java.ClassName;
+import io.spine.tools.java.code.field.FieldFactory;
 import io.spine.tools.mc.java.protoc.ClassMember;
 import io.spine.tools.mc.java.protoc.CodeGenerationTask;
 import io.spine.tools.mc.java.protoc.CompilerOutput;
+import io.spine.tools.protoc.JavaClassName;
 import io.spine.type.MessageType;
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.spine.util.Preconditions2.checkNotEmptyOrBlank;
 
 /**
  * An abstract base for strongly-typed field generation tasks.
  */
 abstract class FieldGenerationTask implements CodeGenerationTask {
 
-    private final ClassName fieldSupertype;
+    private final JavaClassName fieldSupertype;
     private final FieldFactory factory;
 
-    FieldGenerationTask(ClassName fieldSupertype, FieldFactory factory) {
+    FieldGenerationTask(JavaClassName fieldSupertype, FieldFactory factory) {
         this.fieldSupertype = fieldSupertype;
         this.factory = factory;
+        checkNotEmptyOrBlank(fieldSupertype.getCanonical());
```

Comment:
Isn't it a bit (two lines) too late to check for a precondition? Why don't we do when the method starts?